cocotb/_py_compat.py

```diff
@@ -60,3 +61,20 @@ def __exit__(self, *excinfo):
 else:
     import collections
     insertion_ordered_dict = collections.OrderedDict
+
+
+# backport of Python 3.9's functools.cache decorator
+if sys.version_info < (3, 9):
+    def cache(f):
+        return lru_cache(maxsize=None)(f)
+else:
+    from functools import cache  # noqa: F401
+
+
+# Emulates Python 3.8's functools.cached_property decorator.
+# Uses lru_cache with LRU functionality, otherwise the cache could hold references to
+# dead objects and grow infinitely.
```

Comment:
I don't understand this comment - where is `lru_cache` used?

Reply:
Whoops, that got removed.